### PR TITLE
Update loft and migrate more cgal operations

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: algorithm/cgal/cgal.cc [hunk 1]
```diff
@@ -2581,45 +2312,317 @@ void print_polygon_with_holes(
...
+      }
+      strip = strip->parent();
+    }
+    return false;
+  }
+
+  template <typename Queue>
+  void AddA(const Polyline& polyline, Queue& queue) {
+    if (nth_a() + 1 > polyline.size()) {
+      // This prevents going further than wrapping around to the start.
```
Comment (BarbourSmith) [inline algorithm/cgal/cgal.cc:2436]: These comments are all really nice to follow along with

File: algorithm/cgal/cgal.cc [hunk 2]
```diff
@@ -2581,45 +2312,317 @@ void print_polygon_with_holes(
...
+  const Point last_;
+  size_t count_;
+  const FT goodness_;
+  Strip* parent_;
+};
+
+class StripComparator {
+ public:
+  bool operator()(const Strip* a, const Strip* b) {
+    return a->goodness() < b->goodness();
```
Comment (BarbourSmith) [inline algorithm/cgal/cgal.cc:2499]: I like the function name here 🙂...it's clear, but also fun

File: algorithm/cgal/cgal_browser.cjs [hunk 3]
```diff
@@ -1,12 +1,12 @@
 
-var Module = (function() {
+var Module = (() => {
```
Comment (BarbourSmith) [inline algorithm/cgal/cgal_browser.cjs:2]: Why the change of style? Just more compact?

File: api/shape/deform.js [hunk 4]
```diff
@@ -11,8 +11,8 @@ export const deform =
         entries.map(({ selection, deformation }) => ({
           selection: selection.toGeometry(),
           deformation: deformation
-            ? deformation.toGeometry().matrix
-            : identityMatrix,
+            ? deformation.toGeometry()
+            : Point().toGeometry,
```
Comment (BarbourSmith) [inline api/shape/deform.js:15]: 👍

File: nb/documentation/interactions_with_geometry/interactions_with_geometry.md [hunk 5]
```diff
@@ -136,7 +136,7 @@ TODO: In the future (a) => rz(a) can be replaced with rz.
 
 ```JavaScript
 Arc(6)
-  .cut(Arc(3))
+  // .cut(Arc(3))
```
Comment (BarbourSmith) [inline nb/documentation/interactions_with_geometry/interactions_with_geometry.md:139]: Why the delete here?

**Review by BarbourSmith — APPROVED** [review]
Looks great to me!

The framing of some of the .pngs is nicer too which is neat.

File: algorithm/cgal/cgal.cc (hunk 2 above)
Comment (pentacular) [reply algorithm/cgal/cgal.cc:2499]: I was using cost, but I wanted a positive rather than negative gradient :)

File: algorithm/cgal/cgal_browser.cjs (hunk 3 above)
Comment (pentacular) [reply algorithm/cgal/cgal_browser.cjs:2]: I think this was automatically generated by emscripten.

File: nb/documentation/interactions_with_geometry/interactions_with_geometry.md (hunk 5 above)
Comment (pentacular) [reply nb/documentation/interactions_with_geometry/interactions_with_geometry.md:139]: Loft doesn't handle holes with this change.

I left the comment to re-introduce it later.

Comment (BarbourSmith) [reply nb/documentation/interactions_with_geometry/interactions_with_geometry.md:139]: Ah right, I forgot about that. That makes sense.
